src/main/java/org/spongepowered/common/mixin/core/command/server/MixinCommandTeleport.java

```diff
@@ -133,7 +136,9 @@ private static void doTeleport(Entity p_189862_0_, CommandBase.CoordinateArg p_1
             double x = p_189862_1_.getAmount();
             double y = p_189862_2_.getAmount();
             double z = p_189862_3_.getAmount();
+            Sponge.getCauseStackManager().addContext(EventContextKeys.TELEPORT_TYPE, TeleportTypes.COMMAND);
```

Comment:
You should wrap this in a cause stack frame and don't remove from the context (as the frame would be closed)